hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2997,12 +3000,17 @@ public ListSnapshotResponse listSnapshot(
     Map<String, String> auditMap = buildAuditMap(volumeName);
     auditMap.put(OzoneConsts.BUCKET, bucketName);
     try {
+      // Updating the volumeName & bucketName in case the bucket is a linked bucket. We need to do this before a
+      // permission check, since linked bucket permissions and source bucket permissions could be different.
+      ResolvedBucket resolvedBucket = resolveBucketLink(Pair.of(volumeName, bucketName), false);
+      auditMap = buildAuditMap(resolvedBucket.realVolume());
```

Comment:
We don't keep audit logs for read operations. It might been copied-pasted from the write API. If agree, we can create a clean jira.

Reply:
yup we can create a followup jira